File: src/main/kotlin/com/dollee/market/dolleeMarket/utills/NanoIdGenerator.java
package com.dollee.market.dolleeMarket.utills;

import java.io.Serial;
import java.io.Serializable;

import com.aventrix.jnanoid.jnanoid.NanoIdUtils;
import org.hibernate.engine.spi.SharedSessionContractImplementor;
import org.hibernate.id.IdentifierGenerator;

public class NanoIdGenerator implements IdentifierGenerator, Serializable {

  @Serial private static final long serialVersionUID = -7858266607422294454L;

  @Override
  public Serializable generate(
      SharedSessionContractImplementor sharedSessionContractImplementor, Object o) {
    return NanoIdUtils.randomNanoId();
  }
}

File: src/main/kotlin/com/dollee/market/dolleeMarket/utills/NanoIdGenerator.kt
package com.dollee.market.dolleeMarket.utills

import de.huxhorn.sulky.ulid.ULID
import org.hibernate.engine.spi.SharedSessionContractImplementor
import org.hibernate.id.IdentifierGenerator
import java.io.Serial
import java.io.Serializable

class NanoIdGenerator : IdentifierGenerator, Serializable {
    override fun generate(
        sharedSessionContractImplementor: SharedSessionContractImplementor?, o: Any?
    ): Serializable {
        return ULID().nextULID()
    }

    companion object {
        @Serial
        private val serialVersionUID = -7858266607422294454L
    }
}
